packages/static-wado-deploy/lib/DeployGroup.mjs

```diff
@@ -40,20 +41,37 @@ class DeployGroup {
    * @params parentDir is the part of the path to include in the upload name
    * @params name is the item to add
    */
-  async store(parentDir = "", name = "") {
+  async store(parentDir = "", name = "", excludeExisting = {}) {
     const fileName = path.join(this.baseDir, parentDir, name);
     const lstat = await fs.promises.lstat(fileName);
```

Comment:
The store operation is going to throw an exception if the lstat fails so whether it throws a catch(e) { throw new Error(...e)  or just throws directly doesn't actually matter.

Reply:
Ok so you are saying is fine to let server be shutdown just because a file is not found or something?

Reply:
Bill I think Rodolfo s concern is related to the dicomweb server shutdown everytime the command mkdicomweb instances runs when a new annotation in saved to the server. At least is my only concern. I don't think that is a normal behavior...

Reply:
The server doesn't shut down when a file isn't found - what happens is that this throws an exception which cancels the individual upload, which marks it as needing another run of the upload in the parent try/catch when it is running as part of notification, or cancels the upload command when run as a single item.  Neither of those touch the behaviour of either of the webserver or SCP components - those are all separate, and the notification server gets restarted each time an upload occurs, and automatically shuts itself off after a certain number of retries, but then other uploads will cause the study to eventually upload.

Reply:
> Bill I think Rodolfo s concern is related to the dicomweb server shutdown everytime the command mkdicomweb instances runs when a new annotation in saved to the server. At least is my only concern. I don't think that is a normal behavior...

@rodrigobasilio2022 - is that what you are actually seeing?  I agree, the notification server shuts itself down after a certain number of retries, but that is intentional - the design is cooperative and is intended to not stay alive forever as that minimizes system resources when not needing upload behaviour.  I do agree that the notification server was causing webserver issues with the previous execSync instead of the new execSpawn (which I appreciate both you and Rodolfo bringing up those issues - that was a good find and works much better now).  This stat command does not contribute to server shutdown AFAIK, because there is already a catch around it that allows for appropriate retries.

Reply:
We just met @rodrigobasilio2022 @wayfarer3130 thomas and Bill explained server will not be killed because he (now) is prevent child process to kill the parent (the one who holds the server). So, closing this topic.